### PR TITLE
P168077871 more n2 tests

### PR DESCRIPTION
**Pull request by hschilling**
Added an optimization problem to the tests. Added some more tests for all the model viewer data that is used. Created separate function for testing the model_viewer_data.

### CODE REVIEW

File: openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py [hunk 1]
```diff
@@ -88,39 +97,75 @@ def tearDown(self):
...
+                             key=lambda x: (x['tgt'], x['src']))
         for conn in connections:
             if 'cycle_arrows' in conn and conn['cycle_arrows']:
                 cycle_arrows = []
                 for src, tgt in conn['cycle_arrows']:
                     cycle_arrows.append(' '.join([pathnames[src], pathnames[tgt]]))
                 conn['cycle_arrows'] = sorted(cycle_arrows)
-        self.assertEqual(len(connections), len(self.expected_conns))
-        for c, ex in zip(connections, self.expected_conns):
+            self.assertEqual(len(connections), len(expected_conns))
```
Comment (swryan) [inline openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: looks like an indentation error..  you don't mean to check this in the loop do you?

File: openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py [hunk 2]
```diff
@@ -138,31 +183,117 @@ def test_model_viewer_has_correct_data_from_sqlite(self):
...
+                    {"input": {"a_disk.a": "a", "a_disk.Area": "Area", "a_disk.rho": "rho", "a_disk.Vu": "Vu"},
+                    "output": {"indeps.a": "a", "indeps.Area": "Area", "indeps.rho": "rho", "indeps.Vu": "Vu",
+                    "a_disk.Vr": "a_disk.Vr", "a_disk.Vd": "a_disk.Vd", "a_disk.Ct": "a_disk.Ct",
+                    "a_disk.thrust": "a_disk.thrust", "a_disk.Cp": "a_disk.Cp",
+                    "a_disk.power": "a_disk.power"}}
+                    """)
+        expected_declare_partials_betz = json.loads("""
+                ["indeps.a > indeps.a", "indeps.Area > indeps.Area", "indeps.rho > indeps.rho", "indeps.Vu > indeps.Vu", "a_disk.Vr > a_disk.a", "a_disk.Vr > a_disk.Vu", "a_disk.Vd > a_disk.a", "a_disk.Ct > a_disk.a", "a_disk.thrust > a_disk.a", "a_disk.thrust > a_disk.Area", "a_disk.thrust > a_disk.rho", "a_disk.thrust > a_disk.Vu", "a_disk.Cp > a_disk.a", "a_disk.power > a_disk.a", "a_disk.power > a_disk.Area", "a_disk.power > a_disk.rho", "a_disk.power > a_disk.Vu", "a_disk.Vr > a_disk.Vr", "a_disk.Vd > a_disk.Vd", "a_disk.Ct > a_disk.Ct", "a_disk.thrust > a_disk.thrust", "a_disk.Cp > a_disk.Cp", "a_disk.power > a_disk.power"]
+        """)
```
Comment (swryan) [inline openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: seems like we could remove a lot of these `json.loads`
I imagine doing all that loadsing really makes the test slow

File: openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py (hunk 1 above)
Comment (hschilling) [reply openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: Good catch. Fixed.

File: openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py (hunk 2 above)
Comment (hschilling) [reply openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: I checked and even the biggest json.loads takes only about 20 microseconds. Doing it this way was convenient when copying and pasting from the actual value.

File: openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py [hunk 3]
```diff
@@ -88,39 +97,75 @@ def tearDown(self):
...
+        self.assertListEqual(sorted(model_viewer_data['declare_partials_list']),
+                             sorted(expected_declare_partials))
+
+        self.assertEqual(model_viewer_data['driver']['name'], expected_driver_name)
+
+        self.assertListEqual( sorted(dv for dv in model_viewer_data['design_vars']), sorted(expected_design_vars_names))
+        self.assertListEqual( sorted(resp for resp in model_viewer_data['responses']), sorted(expected_responses_names))
+
+    def test_model_viewer_has_correct_data_from_problem(self):
+        """
```
Comment (Kenneth-T-Moore) [inline openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: I remember that there is a weird interaction with testflo (or maybe unittest) and 3-quote docstrings, where the title gets replaced by the docstring in the output.  That's why we (or I) usually use "#" single line comments.

Comment (hschilling) [reply openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: Thanks. I changed those. But there are probably 100 more of those in our tests. Should we make a story to change them?

Comment (Kenneth-T-Moore) [reply openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: Oh, if there are that many, maybe I am wrong and they (bret or unittest devs) have fixed this.

Comment (hschilling) [reply openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: I do remember an issue like that also but it must be fixed.

File: openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py [hunk 4]
```diff
@@ -138,31 +183,117 @@ def test_model_viewer_has_correct_data_from_sqlite(self):
...
+            self.expected_driver_name,
+            self.expected_design_vars_names,
+            self.expected_responses_names
+        )
+
+
+    def test_model_viewer_has_correct_data_from_optimization_problem(self):
+        """
+        Verify that the correct model structure data exists when stored as compared
+        to the expected structure, using the SellarStateConnection model.
```
Comment (swryan) [inline openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: this docstring is actually incorrect (copy/paste error).. it's the ActuatorDisc in this case

Comment (hschilling) [reply openmdao/visualization/n2_viewer/tests/test_viewmodeldata.py]: fixed. Thanks

**Pull request merged**